HW1/P5/P5.txt

```diff
@@ -1 +1,144 @@
+[u'Harvard_University', u'2008', u'2008_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'Legally_Blonde', u'2008_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'2006', u'2006_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'Tommy_Lee_Jones', u'Screen_Actors_Guild_Award_for_Outstanding_Performance_by_a_Cast_in_a_Motion_Picture', u'Kevin_Bacon'],
+[u'Harvard_University', u'The_Yale_Record', u'Mad_(magazine)', u'Kevin_Bacon'],
+[u'Harvard_University', u'Northwestern_University', u'Metra', u'Kevin_Bacon'],
+[u'Harvard_University', u'New_England', u'Litchfield_Hills', u'Kevin_Bacon'],
+[u'Harvard_University', u'Audioslave', u'Elvis_Costello', u'Kevin_Bacon'],
+[u'Harvard_University', u'London', u'Elvis_Costello', u'Kevin_Bacon'],
+[u'Harvard_University', u'2003', u'2003_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'Legally_Blonde', u'2003_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'Pine_Manor_College', u'Dennis_Lehane', u'Kevin_Bacon'],
+[u'Harvard_University', u'Columbia_University', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'Hasty_Pudding_Man_of_the_Year', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'Jack_Lemmon', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'John_Lithgow', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'Hasty_Pudding_Theatricals', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'Tommy_Lee_Jones', u'Dustin_Hoffman', u'Kevin_Bacon'],
+[u'Harvard_University', u'University_of_Virginia', u'Tiki_Barber', u'Kevin_Bacon'],
+[u'Harvard_University', u'University_of_Hartford', u'Apollo_13', u'Kevin_Bacon'],
+[u'Harvard_University', u'Richard_Nixon', u'Apollo_13', u'Kevin_Bacon'],
+[u'Harvard_University', u'Harvard_Man', u'Daniel_Bigel', u'Kevin_Bacon'],
+[u'Harvard_University', u'Natalie_Portman', u'New_York,_I_Love_You', u'Kevin_Bacon'],
+[u'Harvard_University', u'University_of_Hartford', u'Jack_Swigert', u'Kevin_Bacon'],
+[u'Harvard_University', u'August_30', u'Jack_Swigert', u'Kevin_Bacon'],
+[u'Harvard_University', u'John_Kerry', u'John_Edwards_presidential_campaign,_2008', u'Kevin_Bacon'],
+[u'Harvard_University', u'September_16', u'Mickey_Rourke', u'Kevin_Bacon'],
+[u'Harvard_University', u'2007', u'2007_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'The_Great_Debaters', u'2007_in_film', u'Kevin_Bacon'],
+[u'Harvard_University', u'2008', u'Brad_Renfro', u'Kevin_Bacon']
+
+
+Number of connected components:
+The algorithm I used:
+0. cnt is an empty list to store component count
+1. pop one node from total nodes
+2. find all touched nodes connected to the node
+3. count the number of counts and add it to cnt
+4. remove these nodes from the total pool
+5. repeat 1-4 until total nodes pool is empty
+I have tested this algorithm on small datasets, and it returns correct result.
+However, it takes more than one hour on AWS, and I don't have enough credit on my account left.
```

Comment:
Did you find any interesting links? It is great that you print all the found paths.